### PR TITLE
ARC: MPU: Add MPUv8 fixes

### PR DESCRIPTION
**Pull request by kokas-a**
Add mpuv8-related fixes that setup alignment and region overlapping

### CODE REVIEW

**Review by evgeniy-paltsev — DISMISSED** [review]

File: arch/arc/core/mpu/Kconfig [hunk 1]
```diff
@@ -32,8 +32,8 @@ config ARC_MPU
 	select SRAM_REGION_PERMISSIONS
 	select ARC_CORE_MPU
 	select THREAD_STACK_INFO
-	select GEN_PRIV_STACKS if ARC_MPU_VER != 4
-	select MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT if ARC_MPU_VER !=4
-	select MPU_REQUIRES_NON_OVERLAPPING_REGIONS if ARC_MPU_VER = 4
+	select GEN_PRIV_STACKS if (ARC_MPU_VER != 4 || ARC_MPU_VER != 8)
```
Comment (ruuddw) [inline arch/arc/core/mpu/Kconfig:35]: isn't this always true? (same for other places with != and ||)

**Review by ruuddw — CHANGES_REQUESTED** [review]

File: arch/arc/core/mpu/Kconfig (hunk 1 above)
Comment (evgeniy-paltsev) [reply arch/arc/core/mpu/Kconfig:35]: Nice catch.

And to make it less error-prone we can move the `!` outside of parentheses .

```suggestion
	select GEN_PRIV_STACKS if !(ARC_MPU_VER = 4 || ARC_MPU_VER = 8)
```

Comment (kokas-a) [reply arch/arc/core/mpu/Kconfig]: Yes, indeed. Stupid mistake =(

File: arch/arc/core/mpu/Kconfig [hunk 2]
```diff
@@ -32,8 +32,8 @@ config ARC_MPU
 	select SRAM_REGION_PERMISSIONS
 	select ARC_CORE_MPU
 	select THREAD_STACK_INFO
-	select GEN_PRIV_STACKS if ARC_MPU_VER != 4
-	select MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT if ARC_MPU_VER !=4
-	select MPU_REQUIRES_NON_OVERLAPPING_REGIONS if ARC_MPU_VER = 4
+	select GEN_PRIV_STACKS if (!(ARC_MPU_VER = 4 || ARC_MPU_VER = 8))
```
Comment (evgeniy-paltsev) [inline arch/arc/core/mpu/Kconfig:35]: We can drop these extra parentheses to make it more readable - there is no need for these parentheses here.

```suggestion
	select GEN_PRIV_STACKS if !(ARC_MPU_VER = 4 || ARC_MPU_VER = 8)
```

**Review by evgeniy-paltsev — APPROVED** [review]

**Review by ruuddw — APPROVED** [review]

**Pull request merged**